.github/workflows/ci.yml

```diff
@@ -7,8 +7,8 @@ on:
   pull_request:
 
 jobs:
-  build:
-    runs-on: ${{ matrix.os }}
+  test-ubuntu:
```

Comment:
I had to rewrite the entire CI config, because shards got stuck on Github action and I couldn't figure out why. There's a lot of duplication right now, but it works better than before.